x/incentives/keeper/distribute.go

```diff
@@ -313,34 +292,93 @@ func (k Keeper) distributeInternal(
 		remainEpochs = gauge.NumEpochsPaidOver - gauge.FilledEpochs
 	}
 
-	for _, lock := range locks {
-		distrCoins := sdk.Coins{}
-		for _, coin := range remainCoins {
-			// distribution amount = gauge_size * denom_lock_amount / (total_denom_lock_amount * remain_epochs)
-			denomLockAmt := lock.Coins.AmountOfNoDenomValidation(denom)
-			amt := coin.Amount.Mul(denomLockAmt).Quo(lockSum.Mul(sdk.NewInt(int64(remainEpochs))))
-			if amt.IsPositive() {
-				newlyDistributedCoin := sdk.Coin{Denom: coin.Denom, Amount: amt}
-				distrCoins = distrCoins.Add(newlyDistributedCoin)
-			}
+	// This is a no lock distribution flow that assumes that we have a pool associated with the gauge.
+	// Currently, this flow is only used for CL pools. Fails if the pool is not found.
+	// Fails if the pool found is not a CL pool.
+	if gauge.DistributeTo.LockQueryType == lockuptypes.NoLock {
+		pool, err := k.GetPoolFromGaugeId(ctx, gauge.Id, gauge.DistributeTo.Duration)
```

Comment:
Note: all of the new logic in this branch is supposed to be new. Please pay attention to emission rate calculations, choosing incentive record start time and incentive record coins